src/logger.ts

```diff
@@ -173,7 +176,23 @@ export class Logger {
 
     const message = `[${request.method}] [middleware: "middleware"] ${request.nextUrl.pathname}`;
 
-    return this.logHttpRequest(LogLevel.info, message, req, {});
+    if (config?.logRequestDetails) {
+      return requestToJSON(request).then((details) => {
+        const newReq = {
+          ...req,
+          details: Array.isArray(config.logRequestDetails)
+            ? (Object.fromEntries(
+                Object.entries(details as RequestJSON).filter(([key]) =>
+                  (config.logRequestDetails as (keyof RequestJSON)[]).includes(key as keyof RequestJSON)
+                )
+              ) as RequestJSON)
+            : details,
+        };
+        return this.logHttpRequest(LogLevel.info, message, newReq, {});
+      }) as TReturn;
+    }
+
+    return this.logHttpRequest(LogLevel.info, message, req, {}) as TReturn;
```

Comment:
To keep backwards compatibility and also be able to read the `body` we need to conditionally return a Promise that needs to be awaited if `logRequestDetails` is passed. Not the most intuitive solution, so happy to hear suggestions. cc @c-ehrlich